transaction/extra/src/tx_out_confirmation_number.rs

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) 2018-2022 The MobileCoin Foundation
+
+use alloc::vec::Vec;
+use mc_crypto_digestible::Digestible;
+use mc_crypto_hashes::{Blake2b256, Digest};
+use mc_crypto_keys::{RistrettoPrivate, RistrettoPublic};
+use mc_crypto_ring_signature::get_tx_out_shared_secret;
+use mc_util_repr_bytes::{
+    derive_prost_message_from_repr_bytes, typenum::U32, GenericArray, ReprBytes,
+};
+use serde::{Deserialize, Serialize};
+
+/// Domain separator for hashing the confirmation number
+pub const TXOUT_CONFIRMATION_NUMBER_DOMAIN_TAG: &str = "mc_tx_out_confirmation_number";
+
+/// A hash of the shared secret used to confirm tx was sent
+#[derive(
+    Clone, Deserialize, Default, Eq, Ord, PartialEq, PartialOrd, Serialize, Debug, Digestible,
+)]
+pub struct TxOutConfirmationNumber([u8; 32]);
+
+impl TxOutConfirmationNumber {
+    /// Copies self into a new Vec.
+    pub fn to_vec(&self) -> Vec<u8> {
+        self.0.to_vec()
+    }
+
+    /// Validate a confirmation number against tx pubkey and view private key
+    pub fn validate(
+        &self,
+        tx_pubkey: &RistrettoPublic,
+        view_private_key: &RistrettoPrivate,
+    ) -> bool {
+        let shared_secret = get_tx_out_shared_secret(view_private_key, tx_pubkey);
+        let calculated_confirmation = TxOutConfirmationNumber::from(&shared_secret);
+        calculated_confirmation == *self
+    }
+}
+
+impl core::convert::AsRef<[u8; 32]> for TxOutConfirmationNumber {
```

Comment:
Future reference: all these traits are part of `core::prelude`, so we shouldn't need to reference them absolutely.

Reply:
thanks, didn't know this